scripts/prepare_heat_demand.py

```diff
@@ -0,0 +1,367 @@
+# coding: utf-8
+r"""
+Inputs
+-------
+in_path1 : str
+    ``raw/weatherdata``: path of input directory with weather data
+in_path2 : str
+    ``raw/distribution_households.csv``: path of input file with household distributions data as .csv
+in_path3 : str
+    ``raw/holidays.csv``: path of input file with holidays of all states in Germany as .csv
+in_path4 : str
+    ``raw/scalars.csv``: path of template for scalar data as .csv
+in_path5 : str
+    ``oemof_b3/schema/timeseries.csv``: path of input file with times series format as .csv
+out_path : str
+    ``results/_resources/load_profile_heat.csv``: path of output file with prepared data as .csv
+
+Outputs
+---------
+pandas.DataFrame
+    with grouped and aggregated data of heat demands.
+    Data is grouped by region, energy source, technology and chp capability and contains
+    net capacity and efficiency.
+
+Description
+-------------
+The script...
+
+"""
+import sys
+import datetime
+import os
+from demandlib import bdew
+import pandas as pd
+import numpy as np
+
+
+import oemof_b3.tools.data_processing as dp
+
+
+def prepare_weather_data():
+    """
+    This function returns a list with weather data to region
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    weather_data_region : list
+        List of weather data file names of region
+    """
+    weather_data_all = os.listdir(in_path1)
+    weather_data_region = weather_data_all.copy()
+
+    for weather_data_file in weather_data_all:
+        if region not in weather_data_file:
+            weather_data_region.remove(weather_data_file)
```

Comment:
Yes, because its order is connected to the order of `years` in the loop over `years` via `index`:

```
path_weather_data = os.path.join(in_path1, weather_data[index])
```

Reply:
Thanks for this cool advice! :) Applied list comprehension with commit https://github.com/rl-institut/oemof-B3/pull/76/commits/4333fd5eb3080c751215e1f4702911daad17536d.

Reply:
> Yes, because its order is connected to the order of `years` in the loop over `years` via `index`:
> 
> ```
> path_weather_data = os.path.join(in_path1, weather_data[index])
> ```

This is very non-robust and should be avoided. It depends on the two datasets having the same index (see my comment below), which you cannot guarantee and therefore not rely on.